fe/fe-core/src/main/java/com/starrocks/common/io/DeepCopy.java

```diff
@@ -43,6 +43,8 @@
     // the param "c" is the implementation class of "dest".
     // And the "dest" class must has method "readFields(DataInput)"
     public static boolean copy(Writable orig, Writable dest, Class c) {
+        // Backup the current MetaContext before assigning a new one.
+        MetaContext oldContext = MetaContext.get();
```

Comment:
Can you show the problem that the origin code causes

Reply:
The original code will set the MetaContext of the current thread to `null`, any subsequent invoking of the following methods will throw NullPointerException:

https://github.com/StarRocks/starrocks/blob/d350c6cf0979eefe05dec25faa72bd141bbafb9a/fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java#L704

https://github.com/StarRocks/starrocks/blob/d350c6cf0979eefe05dec25faa72bd141bbafb9a/fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java#L700

A specific case of failure that I know is the following unit test:
https://github.com/StarRocks/starrocks/blob/d350c6cf0979eefe05dec25faa72bd141bbafb9a/fe/fe-core/src/test/java/com/starrocks/catalog/TempPartitionTest.java#L203